db/change_cache.go

```diff
@@ -167,7 +167,7 @@ func (c *changeCache) Init(logCtx context.Context, dbcontext *DatabaseContext, n
 	c.initTime = time.Now()
 	c.skippedSeqs = NewSkippedSequenceList()
 	c.lastAddPendingTime = time.Now().UnixNano()
-	c.sgCfgPrefix = base.SGCfgPrefixWithGroupID(dbcontext.Options.GroupID)
+	c.sgCfgPrefix = base.SGCfgPrefixWithGroupID(groupID)
```

Comment:
In this case I might prefer adding a DatabaseCollection.groupID() helper function that traverses back through the dbContext to get the groupID, what do you think?

Reply:
I like this idea, because you don't pass empty string in tests and we already have the value in another attr we pass in.